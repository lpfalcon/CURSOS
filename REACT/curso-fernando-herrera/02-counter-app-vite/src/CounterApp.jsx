import {useState} from 'react'
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
const [counter, setCounter] = useState(value)
    const handledAdd = () => {
        setCounter( counter + 1)
        // setCounter((actualValue)=> actualValue + 1) // otra forma de utilizar el hook useState
    }
    
    const handleSubstract= ()=> setCounter(counter - 1)
    
    
    const handleReset = () => setCounter(value)
    return (
        <main>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={ handledAdd }> +1</button>
            <button onClick={ handleSubstract }> -1</button>
            <button aria-label='btn-reset' onClick={ handleReset }> Reset</button>

        </main>

    )

}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
export default CounterApp