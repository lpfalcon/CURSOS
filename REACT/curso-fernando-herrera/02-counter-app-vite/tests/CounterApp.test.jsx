import { fireEvent, render, screen } from '@testing-library/react'
import CounterApp from '../src/CounterApp'


describe('<CounterApp /> page testing', () => {
    const defaultValue       = 100;
    const stringDefaultValue = '100';
    test('Should match with the snapshot ',() => {

        const { container } =  render(< CounterApp value={defaultValue} />);
        expect( container ).toMatchSnapshot();

    })

    test('Should show the inital value sent in props', () => {

        render(< CounterApp value={defaultValue} />);
        expect( screen.getByText( defaultValue ) ).toBeTruthy();
        expect( screen.getByRole('heading', { level: 2} ).innerHTML)
                .toContain( stringDefaultValue );

    })


    test('Should increment a value when user click +1 button', () => {

        render(< CounterApp value={defaultValue} />);
        fireEvent.click( screen.getByText('+1') );
        expect(screen.getByText(`${defaultValue + 1}`)).toBeTruthy();

    })


    
    test('Should decrement a value when user click +1 button', () => {

        render(< CounterApp value={defaultValue} />);
        fireEvent.click( screen.getByText('-1') );
        // screen.debug();
        expect(screen.getByText(`${defaultValue - 1}`)).toBeTruthy();

    })

    test('Should works reset button', () => {

        render(< CounterApp value={defaultValue} />);
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByRole('button', { name: 'btn-reset'}) );
        expect(screen.getByText(defaultValue)).toBeTruthy();

    })
})