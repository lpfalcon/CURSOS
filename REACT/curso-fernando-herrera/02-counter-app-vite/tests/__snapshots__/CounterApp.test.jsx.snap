// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp /> page testing Should match with the snapshot  1`] = `
<div>
  <main>
    <h1>
      CounterApp
    </h1>
    <h2>
      100
    </h2>
    <button>
       +1
    </button>
    <button>
       -1
    </button>
    <button
      aria-label="btn-reset"
    >
       Reset
    </button>
  </main>
</div>
`;
